src/components/Login.jsx

```diff
@@ -24,23 +24,34 @@ function login(user) {
         .then((data) => data.json());
   }
 }
+const validationSchema = Yup.object({
+  email: Yup
+      .string()
+      .email('Invalid email format')
+      .required('Email is required'),
+  password: Yup
+      .string()
+      .required('Password is required'),
+});
 
 const Login = ({setIsAuthenticated}) => {
   const navigate = useNavigate();
   const cookies = new Cookies();
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-  const [message, setMessage] = useState('');
-  const [showMessage, setShowMessage] = useState(false);
   const [loading, setLoading] = useState(false);
+  const [showMessage, setShowMessage] = useState(false);
+  const [message, setMessage] = useState('');
 
-
-  const handleSubmit = async () => {
-    setLoading(true);
-    const response = await login({email, password});
-    setLoading(false);
-    console.log(response);
-    if (response) {
+  const formik = useFormik({
```

Comment:
Good start! 

`useFormik` is "not meant for the majority of use cases" https://formik.org/docs/api/useFormik

To take full advantage of Formik's API, you should use its [components](https://formik.org/docs/api/formik), which will abstract away most of the boilerplate you have below..

here's how I would write it.

```js
import { Formik, Form, Field } from 'formik';

<Formik
    initialValues={validationSchema.default()}
    validationSchema={validationSchema}
    onSubmit={ async (values) => { ... } }
 >
    {props => (
        <Form>
            <Field name="email" component={TextField} />
            <Field name="password" type="password" component={TextField} />
            <Button>Login</Button>
        </Form>
      )}
 </Formik>
 ```
Look how much less code that is!

Reply:
what about displaying the related error messages

Reply:
https://formik.org/docs/api/errormessage